src/gosling-brush/linear-brush-model.ts

```diff
@@ -0,0 +1,238 @@
+import * as d3Selection from 'd3-selection';
+import { drag as d3Drag } from 'd3-drag';
```

Comment:
To make it clear that these imports are only used for types annotations (and the implementation is provided by HiGlass), I'd recommend using `import type` syntax and capitalizing the imports:

```typescript
import type * as D3Selection from 'd3-selection';
import type * as D3Drag from 'd3-drag';
```

The main issue I see with using's HiGlass's internals (not mentioned in #702) is that we need to make sure our version of our d3 types is identical to HiGlass. This could be challenging to ensure, but we are fortunate with how stable the d3 apis are now.